model/proxy/alphav1/config/cfg.proto

```diff
@@ -21,230 +21,162 @@ package istio.proxy.v1alpha.config;
+message ProxyMeshConfig {
 }
 
-// Describes rules for routing a request/connection to an upstream
-// service, based on attributes associated with the request/connection
-// and downstream service invoking the API call. The choice of the specific
-// upstream cluster will be determined by the routing rule.
-message RouteRule {
-  // Uniquely identifies the upstream service (not its versions) associated
-  // with this RouteRule. This is applicable for both Http and Tcp.
-  // Note: The name used here must match with the 'name' field in the
-  // ClusterIdentifier, that is used to uniquely identify a version of the
-  // upstream service.
+// Destination declares policies that determine how to handle traffic for a
+// destination service (load balancing policies, failure recovery policies such
+// as timeouts, retries, circuit breakers, etc).  Policies are applicable per
+// individual service versions. It is an error to define multiple policies for
+// the same service version.
+//
+// N.B. The policies are enforced on client-side connections or requests, i.e.,
+// enforced when the service is opening a connection/sending a request via the
+// proxy to the destination.
+//
+// N.B. We may extend destination policies to apply to individual service
+// versions and not to the entire service. Proxy mesh may support only a subset
+// of policies to be applicable to individual service version destinations.
+message Destination {
+  // Service for which the service version is defined.
   string destination = 1;
-  oneof route_rule {
-    L4RouteRule layer4 = 2;
-    HttpRouteRule http = 3;
-  }
-}
 
-// Routes incoming Tcp/Udp traffic to one of the clusters of a service
-// based on match criterion.
-message L4RouteRule {
-  // Set of conditions that must be satisfied, such as downstream cluster
-  // labels, connection attributes, etc.
-  L4MatchCondition match = 1;
-
-  // Each routing rule is associated with one or more upstream clusters,
-  // (see glossary in beginning of document). Weights associated with the
-  // cluster determine the proportion of traffic it receives.
-  repeated WeightedCluster weighted_clusters = 2;
-
-  // Faults can be injected into the connections from downstream by the
-  // proxy, for testing the failure recovery capabilities of downstream
-  // services.  Faults include aborting the request/connection from
-  // downstream service, delaying the proxying of request/connection to the
-  // upstream service, and throttling the bandwidth of the connection
-  // (either end).
-
-  // TODO: This should be done on per-cluster basis.
-  L4FaultInjection fault = 3;
-}
+  // Load balancing policy
+  LoadBalancing load_balancing = 2;
+
+  // Timeout policy
+  Timeout timeout = 3;
+
+  // Retry policy
+  Retry retry = 4;
+
+  // Circuit breaker policy
+  CircuitBreaker circuit_breaker = 5;
+
+  // L7 fault injection policy applies to L7 traffic
+  HTTPFaultInjection http_fault = 6;
```

Comment:
We don't need oneof constraint since the fault injection policies don't apply to the same traffic. If you have L4 service, you write L4 fault policy; if you have both L4 and L7 ports, you write both fault policies?